core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -141,7 +141,8 @@ class GroupMetadataManager(val brokerId: Int,
       // if we crash or leaders move) since the new leaders will still expire the consumers with heartbeat and
       // retry removing this group.
       val groupPartition = partitionFor(group.groupId)
-      val tombstone = new Message(bytes = null, key = GroupMetadataManager.groupMetadataKey(group.groupId))
+      val tombstone = new Message(bytes = null, key = GroupMetadataManager.groupMetadataKey(group.groupId),
+        timestamp = SystemTime.milliseconds, magicValue = Message.MagicValue_V1)
```

Comment:
Yes, you are right. Didn't see the message format check in KafkaConfig. I think it's still better and simpler to just use the message format specified for the internal topic. This way, one can wait until inter protocol is upgraded to 0.10.0 in all brokers and then upgrade the message format w/o any conversion overhead. If we rely upon inter-broker protocol, there will still be some conversion while changing the inter-broker protocol in all brokers.